contracts/src/PDPRecordKeeper.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+// PDPRecordKeeper is a default implementation of a record keeper for the PDP service.
+// It maintains a record of all events that have occurred in the PDP service,
+// and provides a way to query these events.
+contract PDPRecordKeeper {
+    // The address of the PDP service contract that is allowed to call this contract
+    address public pdpServiceAddress;
+
+    // Enum to represent different types of operations
+    enum OperationType {
+        CREATE,
+        ADD,
+        REMOVE,
+        PROVE_POSSESSION,
+        DELETE
+    }
+
+    // Struct to store event details
+    struct EventRecord {
+        uint64 epoch;
+        OperationType operationType;
+        bytes extraData;
+    }
+
+    // Mapping to store events for each proof set
+    mapping(uint256 => EventRecord[]) public proofSetEvents;
+
+    // Event emitted when a new record is added
+    event RecordAdded(uint256 indexed proofSetId, uint64 epoch, OperationType operationType);
+
+    constructor(address _pdpServiceAddress) {
+        pdpServiceAddress = _pdpServiceAddress;
```

Comment:
This will shake out more in integration but the current deploy flow I have planned is 
1) deploy PDP service
2) deploy PDP record keeper with service address
3) proof set create will now require record keeping address and proof set create wont work if there is no contract setup to receive events

Reply:
And if we move to upgradable PDP service during test we'll also make this default record keeper upgradeable